streamz/core.py

```diff
@@ -1047,18 +1056,33 @@ class unique(Stream):
     3
     """
     def __init__(self, upstream, history=None, key=identity, **kwargs):
-        self.seen = dict()
+        self.seen = None
         self.key = key
-        if history:
-            from zict import LRU
-            self.seen = LRU(history, self.seen)
+        self.history = history
 
         Stream.__init__(self, upstream, **kwargs)
 
     def update(self, x, who=None):
         y = self.key(x)
+        # If this is the first piece of data make the cache
+        if self.seen is None:
+            if isinstance(y, Hashable):
+                self.seen = dict()
+                if self.history:
+                    # if it is hashable use LRU cache
+                    if isinstance(y, Hashable):
```

Comment:
This seems to be an `and`, and I would say that the latter should always be True